pydoctor/test/test_epydoc2stan.py

```diff
@@ -445,3 +446,52 @@ def test_xref_not_found_restructured(capsys: CapSys) -> None:
     #       the line number when it calls visit_title_reference().
     #       https://github.com/twisted/pydoctor/issues/237
     assert captured == "test:3: invalid ref to 'NoSuchName' not resolved\n"
+
+
+class RecordingAnnotationLinker(DocstringLinker):
+    """A DocstringLinker implementation that cannot find any links,
+    but does record which identifiers it was asked to link.
+    """
+
+    def __init__(self) -> None:
+        self.requests: List[str] = []
+
+    def resolve_identifier(self, identifier: str) -> Optional[str]:
+        self.requests.append(identifier)
+        return None
+
+    def resolve_identifier_xref(self, identifier: str, lineno: int) -> str:
+        assert False
+
+@mark.parametrize('annotation', (
+    '<bool>',
```

Comment:
minor comment:

to make it easy to read and reduce the places were errors can be introduced, when creating parametrized test method I prefer to pass the input and the exact (already resolved) expected value.
So in this case, replace the `expected_lookups` and `expected_text` with actual values passes to `@mark.parametrize`

Reply:
I actually had it like that at first, but I thought the compact syntax was more readable. For example:
```py
    '<Callable>[[<str>, <bool>], <None>]',
vs
    ('Callable[[str, bool], None]', ('Callable', 'str', 'bool', 'None')),
```
It does make the test code slightly more complex, but it only adds 2 lines so it's not that big of a risk.

Note that depending on what we pick in #280, the rule that the expected text must be equal to the input may no longer hold. If so, we'd have to switch to a 3-parameter format: input, expected text and expected lookups.